saleor/tests/e2e/orders/test_order_product_with_percentage_promotion.py

```diff
@@ -0,0 +1,240 @@
+        channel_slug,
+        variant_price,
+        promotion_name,
+        discount_value,
+        discount_type,
+        promotion_rule_name,
+    )
+
+    # Step 1 - Create a draft order for a product with fixed promotion
+    input = {
+        "channelId": channel_id,
+        "billingAddress": DEFAULT_ADDRESS,
+        "shippingAddress": DEFAULT_ADDRESS,
+        "shippingMethod": shipping_method_id,
+    }
+    data = draft_order_create(e2e_staff_api_client, input)
+    order_id = data["order"]["id"]
+    assert data["order"]["billingAddress"] is not None
+    assert data["order"]["shippingAddress"] is not None
+    assert order_id is not None
+
+    # Step 2 - Add order lines to the order
+    lines = [{"variantId": product_variant_id, "quantity": 1}]
+    order_lines = order_lines_create(e2e_staff_api_client, order_id, lines)
+    order_product_variant_id = order_lines["order"]["lines"][0]["variant"]["id"]
+    discount = round(float(variant_price) * discount_value / 100, 2)
+    assert order_product_variant_id == product_variant_id
+    unit_price = float(variant_price) - float(discount)
+    undiscounted_price = order_lines["order"]["lines"][0]["undiscountedUnitPrice"][
+        "gross"
+    ]["amount"]
+    assert float(undiscounted_price) == float(variant_price)
+    assert (
+        order_lines["order"]["lines"][0]["unitPrice"]["gross"]["amount"] == unit_price
+    )
+    promotion_reason = order_lines["order"]["lines"][0]["unitDiscountReason"]
+    assert (
+        promotion_reason
+        == f"Promotion rules discounts: {promotion_name}: {promotion_rule_name}"
+    )
+
+    # Step 3 - Add a shipping method to the order
+    input = {"shippingMethod": shipping_method_id}
+    draft_update = draft_order_update(e2e_staff_api_client, order_id, input)
+    order_shipping_id = draft_update["order"]["deliveryMethod"]["id"]
+    shipping_price = draft_update["order"]["shippingPrice"]["gross"]["amount"]
+    subtotal_gross_amount = draft_update["order"]["subtotal"]["gross"]["amount"]
+    total_gross_amount = draft_update["order"]["total"]["gross"]["amount"]
+    assert order_shipping_id is not None
+
+    # Step 4 - Complete the draft order
+    order = draft_order_complete(e2e_staff_api_client, order_id)
+    order_complete_id = order["order"]["id"]
+    assert order_complete_id == order_id
+    order_line = order["order"]["lines"][0]
+    assert order_line["productVariantId"] == product_variant_id
+    assert order_line["unitDiscount"]["amount"] == float(discount)
+    assert order_line["unitDiscountType"] is not None
+    assert order_line["unitDiscountValue"] == float(discount)
+    assert order_line["unitDiscountReason"] == promotion_reason
+    product_price = order_line["undiscountedUnitPrice"]["gross"]["amount"]
+    assert product_price == float(undiscounted_price)
+    assert product_price == float(variant_price)
+    total = order["order"]["total"]["gross"]["amount"]
+    assert total == float(total_gross_amount)
```

Comment:
Here we are comparing the `total` with the `total` from the previous step. IMO we should calculate this value and check if it's correct.
Same for `subtotal`.